driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/open.bpf.c

```diff
@@ -73,7 +73,9 @@ int BPF_PROG(open_x,
 
 	/* Parameter 3: flags (type: PT_FLAGS32) */
 	u32 flags = (u32)extract__syscall_argument(regs, 1);
-	auxmap__store_u32_param(auxmap, open_flags_to_scap(flags));
+	u32 scap_flags = (u32)open_flags_to_scap(flags);
+	scap_flags |= extract__fmode_created_from_fd(ret);
+	auxmap__store_u32_param(auxmap, scap_flags);
```

Comment:
This is the reason why e2e tests are failing, `flags` is reused in the `open_modes_to_scap` method. We need to extend this fix to `kmod` and other syscalls